File: BOJ 2/2799.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
int main() {
    int n,m;
    scanf("%d%d",&n,&m);
    char **board = new char*[5*n+1];
    for(int i=0;i<5*n+1;i++) board[i] = new char[5*m+1];
    for(int i=0;i<5*n+1;i++){
        for(int j=0;j<5*m+1;j++){
            scanf(" %c",&board[i][j]);
        }
    }
    int zero =0,one =0,two =0,three =0,four = 0;
    for(int i=1,k=0;k<n;i+=5,k++){
        for(int j=1,p=0;p<m;j+=5,p++){
            int star = 0;
            for(int a=0;a<4;a++){
                for(int b=0;b<4;b++){
                    if(board[i+a][j+b] == '*') star++;
                }
            }
            if(star ==0) zero++;
            else if(star == 4) one++;
            else if(star == 8) two++;
            else if(star == 12) three++;
            else four++;
        }
    }
    printf("%d %d %d %d %d\n",zero,one,two,three,four);
    for(int i=0;i<5*n+1;i++) delete[] board[i];
    delete[] board;
    return 0;
}

File: BOJ(Java 공부용)/2799.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 0, 1, -1};
    static int[] goy = {1, -1, 0, 0};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken());
        char[][] board = new char[5 * n + 1][5 * m + 1];
        for (int i = 0; i < 5 * n + 1; i++) {
            String s = br.readLine();
            for (int j = 0; j < s.length(); j++) {
                if (s.charAt(j) == ' ') break;
                board[i][j] = s.charAt(j);
            }
        }
        int one = 0, two = 0, three = 0, four = 0, five = 0;
        for (int i = 1; i < 5 * n + 1; i += 5) {
            for (int j = 1; j < 5 * m + 1; j += 5) {
                int res = go(i, j, board);
                if (res == 0) one++;
                if (res == 4) two++;
                if (res == 8) three++;
                if (res == 12) four++;
                if (res == 16) five++;
            }
        }

        bw.write(one + " " + two + " " + three + " " + four + " " + five);
        bw.flush();
        bw.close();
    }

    private static int go(int x, int y, char[][] board) {
        int count = 0;
        for (int i = x; i < x + 4; i++) {
            for (int j = y; j < y + 4; j++) {
                if (board[i][j] == '*') count++;
            }
        }
        return count;
    }
}